api/src/main/java/com/velocitypowered/api/proxy/Player.java

```diff
@@ -275,4 +283,36 @@ default void sendMessage(net.kyori.text.Component component) {
    */
   @Override
   boolean sendPluginMessage(ChannelIdentifier identifier, byte[] data);
+
+  @Override
+  default @NotNull Pointers pointers() {
+    return CommandSource.super.pointers().toBuilder()
+            .withDynamic(Identity.UUID, this::getUniqueId)
+            .withDynamic(Identity.NAME, this::getUsername)
+            .withDynamic(Identity.DISPLAY_NAME, this::getDisplayName)
+            .build();
+  }
```

Comment:
This should be cached in the implementation class instead.

Reply:
permissions as well?

Reply:
yes

Reply:
thank you, done :)